server/src/api/community.rs

```diff
@@ -652,6 +652,16 @@ impl Perform for Oper<BanFromCommunity> {
 
     let user_id = claims.id;
 
+    let community_id = data.community_id;
+    let community_moderators = blocking(pool, move |conn| {
+      CommunityModeratorView::for_community(conn, community_id)
+    })
+    .await??;
+
+    if !community_moderators.iter().any(|m| m.user_id == user_id) {
+      return Err(APIError::err("couldnt_update_community").into());
+    }
+
```

Comment:
These also need to add the admins (admins can appoint people mods even for communities they don't control).

From `comment.rs`: 
```
      editors.append(
        &mut blocking(pool, move |conn| {
          Ok(UserView::admins(conn)?.into_iter().map(|a| a.id).collect()) as Result<_, LemmyError>
        })
        .await??,
      );
```

Reply:
fixed